app/src/main/java/com/rahobbs/todo/fragments/TodoFragment.java

```diff
@@ -0,0 +1,228 @@
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.client.Firebase;
+import com.rahobbs.todo.helpers.Feedback;
+import com.rahobbs.todo.R;
+import com.rahobbs.todo.helpers.TodoItem;
+import com.rahobbs.todo.helpers.TodoLab;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.UUID;
+
+/**
+ * Created by rachael on 2/1/16.
+ * Fragment that contains a single to-do item detailed view
+ */
+public class TodoFragment extends Fragment {
+
+    private static final String ARG_TODO_ID = "todoId";
+    private static final String DIALOG_DATE = "DialogDate";
+    private static final int REQUEST_DATE = 0;
+
+    private TodoItem mTodo;
+    private EditText mTitle;
+    private EditText mDetails;
+    private TextView mDueDateTextField;
+    private CheckBox mCompletedCheckbox;
+    private LinearLayout mDateComponents;
+
+    public static TodoFragment newInstance(UUID todoId) {
+        Bundle args = new Bundle();
+        args.putSerializable(ARG_TODO_ID, todoId);
+        TodoFragment fragment = new TodoFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Firebase.setAndroidContext(getContext());
+        UUID todoId = (UUID) getArguments().getSerializable(ARG_TODO_ID);
+        setHasOptionsMenu(true);
+
+        if (todoId == null) {
+            throw new RuntimeException("Todo ID is null.");
+        }
+
+        mTodo = TodoLab.get(getActivity()).getItem(todoId);
+        Log.v("DETAIL_CREATED", mTodo.getID().toString());
+
+        if (mTodo == null) {
+            throw new RuntimeException("No Todo item with provided ID (" + todoId.toString() + ") was found.");
+        }
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.fragment_todo_item, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_send_feedback:
+                Intent i = new Intent(Intent.ACTION_SEND);
+                Feedback fb = new Feedback();
+                fb.sendFeedback(i);
+                startActivity(Intent.createChooser(i, "Send mail..."));
+                return true;
+            case R.id.menu_item_delete_todo:
+                UUID todoId = mTodo.getID();
```

Comment:
there is soooo much happening in this one case. Whenever you look at code and see a ton of different things happening, that's usually a sign to stop and ask, "what can i pull out of this".

In this case it looks like three things are happening:
1. deleting the `TodoItem`
2. Create a title for a toast
3. Creating a a toast

I would pull out everything relating to the toast creation into a separate method. It'll make this a lot easier to read. Also you're finishing the activity right after you call show toast, you want to make sure a user actually has a chance to see your toast, so you might want to play around with delaying the finish call.